src/components/banner-component.jsx

```diff
@@ -0,0 +1,65 @@
+import classNames from "classnames";
+import { Show, onCleanup } from "solid-js";
+import { Button } from ".";
+import { banner_category } from "../constants/banner-category";
+import { setBannerMessage } from "../stores";
+import styles from "../styles/banner-component.module.scss";
+import { isDesktop } from "../utils/responsive";
+
+const OverlayWrapper = ({ condition, wrapper, children }) =>
+  condition ? wrapper(children) : children;
+
+const BannerComponent = (props) => {
+  const nav_height = document.getElementById("app_navbar")?.offsetHeight;
+
+  onCleanup(() => {
+    setBannerMessage(null);
+  });
+  return (
+    <Show when={props.message}>
+      <OverlayWrapper
+        condition={isDesktop()}
+        wrapper={(children) => (
+          <div
+            class={styles["popup__div"]}
+            onClick={() => setBannerMessage(null)}
+          >
+            {children}
+          </div>
+        )}
+      >
+        <div
+          style={{ top: nav_height }}
+          class={`${classNames(styles.popup, {
```

Comment:
While using classNames() u need not interpolate the expression. the resultant value will be string

Reply:
removed interpolation and updated the pr